### PR TITLE
docs(agent): clarify Agent components

### PR DESCRIPTION
**Pull request by aimeeu**
Clarify that there is a Clouddriver plugin and a separate Agent
service that runs in the K8s cluster.

Resolves Jira: [DOC-433]

Deploy preview:  https://deploy-preview-749--armory-docs.netlify.app/docs/armory-agent/

[DOC-433]: https://armory.atlassian.net/browse/DOC-433

### CODE REVIEW

File: content/en/includes/agent/agent-compat-matrix.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 > The Armory Agent is in early access. For more information about using this feature, [contact us](https://www.armory.io/contact-us/).
 
-The Armory Agent is compatible with Armory Enterprise and open source Spinnaker. It consists of a lightweight service that you deploy on Kubernetes and a plugin that you install into Spinnaker. Your Clouddriver service must use a MySQL-compatible database. See the {{< linkWithTitle "clouddriver-sql-configure.md" >}} guide for instructions.
+The Armory Agent is compatible with Armory Enterprise and open source Spinnaker. It consists of a lightweight Agent service that you deploy on Kubernetes and a plugin that you install into Clouddriver. Your Clouddriver service must use a MySQL-compatible database. See the {{< linkWithTitle "clouddriver-sql-configure.md" >}} guide for instructions.
```
Comment (317brian) [inline content/en/includes/agent/agent-compat-matrix.md]: I'd expect a sentence like this to be near the start of the main Agent page 

```
It consists of a lightweight Agent service that you deploy on Kubernetes and a plugin that you install into Clouddriver.

File: content/en/docs/armory-agent/_index.md [hunk 2]
```diff
@@ -19,85 +19,86 @@ description: >
...
+  * Use the Agent alongside Armory Enterprise and benefit from performance improvements.
   * Use the Agent in a target cluster and get Kubernetes accounts automatically registered.
   * Use YAML, a HELM chart, or a Kustomize template to inject the Agent into newly provisioned Kubernetes clusters, and immediately make those clusters software deployment targets.
-  * Use the Agent with little operational overhead or changes in the way you currently manage Spinnaker.
+  * Use the Agent with little operational overhead or changes in the way you currently manage Armory Enterprise.
 
-Check out the Quick Start [guide]({{< ref "armory-agent-quick" >}}) to deploy the Agent on your Kubernetes infrastructure.
+Check out the installation [guide]({{< ref "armory-agent-quick" >}}) for how to deploy the Agent components in Armory Enterprise and in your Kubernetes infrastructure.
 
 ## Deployment topologies
```
Comment (317brian) [inline content/en/docs/armory-agent/_index.md]: Is it just me or is this a really odd/non-descriptive title for this section? I'm thrown off by the word topology.

Maybe something like `Deployment modes (or options)`?

**Review by 317brian — CHANGES_REQUESTED** [review]
Minor comments but otherwise the changes are very helpful. No more confusion about "Agent" since they're both actually parts of the Agent.

File: content/en/includes/agent/agent-compat-matrix.md (hunk 1 above)
Comment (aimeeu) [reply content/en/includes/agent/agent-compat-matrix.md]: Agree. Moved out of the compat include and onto the index page.

**Review by 317brian — APPROVED** [review]
🎉

**Pull request merged**